scrunch/expressions.py

```diff
@@ -433,36 +433,60 @@ def _process(obj, variables):
                 subvariables = arrays[0]
                 value = values[0]
 
-                if op == 'all':
-                    if len(value['value']) != 1:
-                        raise ValueError
-                    value['value'] = value['value'][0]
+                def _extract_id(url):
+                    '''Extracts the id form the given url'''
+                    return [
+                        part for part in url.split('/') if part != ''
+                    ][-1]
+
+                subvariables_dict = {
+                    _extract_id(subvar): subvar
+                    for subvar in subvariables
+                }
+
+                # no expansion is needed if all arguments are subvariables,
+                # instead we should just transform variable references to
+                # their corresponding urls
+                are_subvars = all(
+                    subvar in value['value']
+                    for subvar in subvariables_dict.keys()
+                )
 
-                if len(subvariables) == 1:
-                    obj['function'] = real_op
-                    obj['args'][0] = {'variable': subvariables[0]}
-                    obj['args'][1] = value
+                if are_subvars:
```

Comment:
I will need to revisit this later, the way we'll be dealing with numeric subvariable responses is by using the following mask `"%s_%s"  % (parent_id, number)`, this will happen in Scrunch under the covers, I'll add the other case later when my branch lands.